integration-tests/app_id_test.go

```diff
@@ -23,15 +26,13 @@ type AppCredentialTestSuite struct {
 }
 
 func (suite *AppCredentialTestSuite) SetupSuite() {
-	err := StartProxy("./", "./envoy.yaml")
-	require.Nilf(suite.T(), err, "Error starting proxy: %v", err)
 	// Initializing 3scale backend state
 	suite.AppID = "test-app-id"
 	suite.AppKey = "test-app-key"
 	suite.UserKey = "test-user-key"
-	suite.ServiceID = "test-service-id"
+	suite.ServiceID = uuid.NewString()
 	suite.PlanID = "test-plan-id"
-	suite.ServiceToken = "test-service-token"
+	suite.ServiceToken = uuid.NewString()
```

Comment:
Since we have serialized tests, I don't think so we should have a case where ITs fail due to duplicate service without any error on the writer side. I was thinking about parallelizing the tests but that would require a different solution than this, which I can discuss if you want.

Reply:
There can be issues in the network or in the server side as well.  Also there's another issue with stats. When you create a usage and delete it the stats won't get deleted. We have to delete the stats which we don't do at the moment. So when you create a new service/app combination, it already contains stats. Managing apisonator apps, services and keys is a useless effort since it doesn't provide us any benefit and all the apisonator states get deleted after the tests. We don't have to worry about the apisonator apps, services when we use UUIDs.

Reply:
Weird, I've only seen running tests in parallel but that may be just my experience.

Reply:
How you design and write tests depends on the requirement. So I'm not saying that parallel tests are bad. I'm saying that for our case it is not feasible to run several proxy instances for the integration tests. apisonator records the usages as stats. You can see the redis database for the stats. We are not deleting the stats. We don't need to consider those stats since we use UUIDs. Otherwise if you create the new service app key combination, it will have the usage from the previous tests.